### PR TITLE
fix dnsblock example, add hooks field

### PR DESCRIPTION
**Pull request by sheharyaar**

### CODE REVIEW

File: examples/dnsblock/dnsblock.lua [hunk 1]
```diff
@@ -68,5 +68,6 @@ xt.match{
 	match = dnsblock_mt,
 	checkentry = nop,
 	destroy = nop,
+	hooks = 0,
```
Comment (lneto) [inline examples/dnsblock/dnsblock.lua:71]: Shouldn't this be optional?

Comment (sheharyaar) [reply examples/dnsblock/dnsblock.lua:71]: It can be optional, but we have not added optional table field checking in luaxtable.

Comment (lneto) [reply examples/dnsblock/dnsblock.lua:71]: what `0` means in this case.. that is, how netfilter will handle that? also, what happens if we pass a invalid value here?

Comment (sheharyaar) [reply examples/dnsblock/dnsblock.lua:71]: 0 means, that it can be added to any hook point that is provided by a particular nat table

Comment (lneto) [reply examples/dnsblock/dnsblock.lua:71]: Isn't there an "any" flag for this? And what happens in case of invalid value?

Comment (sheharyaar) [reply examples/dnsblock/dnsblock.lua:71]: The value is checked against a mask, and if it is not valid, `EINVAL` is returned.

Comment (sheharyaar) [reply examples/dnsblock/dnsblock.lua:71]:
```c
	if (par->target->hooks && (par->hook_mask & ~par->target->hooks) != 0) {
		char used[64], allow[64];

		pr_info_ratelimited("%s_tables: %s target: used from hooks %s, but only usable from %s\n",
				    xt_prefix[par->family], par->target->name,
				    textify_hooks(used, sizeof(used),
						  par->hook_mask, par->family),
				    textify_hooks(allow, sizeof(allow),
						  par->target->hooks,
						  par->family));
		return -EINVAL;
	}
```

**Pull request merged**